exasol_advanced_analytics_framework/udf_communication/peer_communicator/acknowledge_register_peer_sender.py

```diff
@@ -56,6 +56,11 @@ def _should_we_send(self):
         result = is_time and not self._finished
         return result
 
+    def is_ready_to_stop(self):
+        result = (self._finished and self._needs_to_send_for_peer) or not self._needs_to_send_for_peer
```

Comment:
as logically the same as
```suggestion
        result = self._finished or not self._needs_to_send_for_peer
```